### PR TITLE
Reduce warnings from shadowing symbols

### PR DESCRIPTION
**Pull request by linus-sherrill**
Generally shadowing symbols is benign, but there are cases where it has masked a legitimate bug. It is not that difficult to use an alternate name to avoid shadowing.

### CODE REVIEW

**Comment by mwoehlke-kitware** [comment]
I think you meant for only 5192f9accec9e64f2a50ddf29c2d3ff272f6280c to be in here?

**Review by mleotta — COMMENTED** [review]
Generally this is okay.  I've point out some cases where I would have done the renaming differently.  These are just suggestions, I'm not going to push back too hard if we want to merge as is.

File: CMake/kwiver-flags-gnu.cmake [hunk 1]
```diff
@@ -16,6 +16,7 @@ kwiver_check_compiler_flag( -Werror=reorder )
 kwiver_check_compiler_flag( -Werror=overloaded-virtual )
 kwiver_check_compiler_flag( -Werror=cast-qual )
 kwiver_check_compiler_flag( -Werror=vla )
+kwiver_check_compiler_flag( -Wshadow=local )
```
Comment (mwoehlke-kitware) [inline CMake/kwiver-flags-gnu.cmake:19]: Thanks :slightly_smiling_face:

File: arrows/ceres/options.cxx [hunk 2]
```diff
@@ -678,17 +678,17 @@ ::add_forward_motion_damping_cost(
     const double scale = constraints.size() / sum_dist;
     auto scaled_loss = new ::ceres::ScaledLoss(NULL, weight,
                            ::ceres::Ownership::TAKE_OWNERSHIP);
-    for (auto curr_cam : constraints)
+    for (auto curr_cam_constr : constraints)
     {
-      auto prev_cam = curr_cam - 1;
-      double inv_dist = 1.0 / static_cast<double>(curr_cam->first -
-                                                  prev_cam->first);
+      auto p_cam = curr_cam_constr - 1;
```
[the remark this answers is not in the document]
Comment (mwoehlke-kitware) [reply arrows/ceres/options.cxx:683]: For me, I'm less a fan of abbreviations :slightly_smiling_face:. Would `curr_constraint` and `prev_constraint` work? ("prev" is sufficiently common I'll let that one slide, especially as it is the same length as "next", and I'll let "curr" go for the same reason. However, "constr" makes me think "constructor". The "cam" seems superfluous?)

File: arrows/core/close_loops_keyframe.cxx [hunk 3]
```diff
@@ -272,14 +272,14 @@ ::stitch( vital::frame_id_t frame_number,
     all_matches[*f] = pool.enqueue(match_func, *f);
   }
   // stitch with all previous keyframes
-  for(auto kitr = keyframes.rbegin(); kitr != keyframes.rend(); ++kitr)
```
Comment (mwoehlke-kitware) [inline arrows/core/close_loops_keyframe.cxx:275]: Given how far apart setting it is to its use, I would have left these alone and instead renamed the first `kitr` to something more meaningful. (Maybe @mleotta has a naming suggestion?)

File: arrows/core/depth_utils.cxx [hunk 4]
```diff
@@ -159,8 +159,8 @@ project_3d_bounds(kwiver::vital::vector_3d const& minpt,
 
   for (vector_3d const& p : points)
   {
-    vector_2d pp = cam.project(p);
-    int ui = static_cast<int>(pp[0]), vi = static_cast<int>(pp[1]);
+    const vector_2d p_p = cam.project(p);
```
[the remark this answers is not in the document]
Comment (mwoehlke-kitware) [reply arrows/core/depth_utils.cxx:162]: I think I would instead rename the previous one `pp0`.

Also, the next change looks fishy, taking the X of `project(points[0])` and the Y of `project(p)`?

+request changes

File: arrows/kpf/yaml/kpf_yaml_parser.cxx [hunk 5]
```diff
@@ -461,9 +461,9 @@ parse_activity( const YAML::Node& n, KPF::packet_buffer_t& local_packet_buffer )
       }
       else if (h.style == KPF::packet_style::EVAL)
       {
-        KPF::packet_t p(h);
-        if ( ! parse_packet( i, p )) return false;
-        act.evals.push_back ( {p.eval, p.header.domain} );
+        KPF::packet_t pp(h);
```
Comment (mwoehlke-kitware) [inline arrows/kpf/yaml/kpf_yaml_parser.cxx:464]: `pp` seems odd; is "packet" available?

File: arrows/mvg/algo/initialize_cameras_landmarks_basic.cxx [hunk 6]
```diff
@@ -1028,8 +1028,8 @@ ::initialize(camera_map_sptr& cameras,
 
     // find existing landmarks for tracks also having features on the other frame
     map_landmark_t flms;
-    std::vector<track_sptr> trks = ftracks->active_tracks(static_cast<int>(other_frame));
-    for(const track_sptr& t : trks)
+    std::vector<track_sptr> ftrks = ftracks->active_tracks(static_cast<int>(other_frame));
```
Comment (mwoehlke-kitware) [inline arrows/mvg/algo/initialize_cameras_landmarks_basic.cxx:1031]: FWIW, I vote for "aftracks" instead...

File: arrows/ocv/feature_detect_extract_ORB.cxx [hunk 7]
```diff
@@ -165,9 +165,9 @@ class priv
     bool valid = true;
 
     // Must be one of the enumeration values
-    int score_type = config->get_value<int>( "score_type" );
-    if( ! ( score_type == cv::ORB::HARRIS_SCORE ||
-            score_type == cv::ORB::FAST_SCORE ) )
+    int temp = config->get_value<int>( "score_type" );
```
Comment (mwoehlke-kitware) [inline arrows/ocv/feature_detect_extract_ORB.cxx:168]: FWIW, I vote for "st" instead...

File: arrows/ocv/match_features_bruteforce.cxx [hunk 8]
```diff
@@ -43,10 +43,10 @@ namespace ocv {
 class match_features_bruteforce::priv
 {
 public:
-  priv(int norm_type=cv::NORM_L2, bool cross_check=false)
-      : norm_type( norm_type ),
-        cross_check( cross_check ),
-        matcher( new cv::BFMatcher(norm_type, cross_check) )
+  priv(int n_type = cv::NORM_L2, bool c_check = false)
```
Comment (mwoehlke-kitware) [inline arrows/ocv/match_features_bruteforce.cxx:46]: Yeah, this is one of those times I'd tell `-Wshadow` to go get bent... not saying I'm requesting that, just saying that's what I'd do.

File: arrows/serialize/protobuf/tests/test_convert_protobuf.cxx [hunk 9]
```diff
@@ -395,7 +395,7 @@ TEST( convert_protobuf, detected_object_set )
       auto ser_it = ser_dot_sptr->begin();
       auto dser_it = dser_dot_sptr->begin();
 
-      for ( size_t i = 0; i < ser_dot_sptr->size(); ++i )
+      for ( size_t idx = 0; idx < ser_dot_sptr->size(); ++idx )
```
Comment (mwoehlke-kitware) [inline arrows/serialize/protobuf/tests/test_convert_protobuf.cxx:398]: "j"?

File: arrows/serialize/protobuf/tests/test_serialize.cxx [hunk 10]
```diff
@@ -424,7 +424,7 @@ TEST( serialize, detected_object_set )
       auto ser_it = ser_dot_sptr->begin();
       auto deser_it = deser_dot_sptr->begin();
 
-      for ( size_t i = 0; i < ser_dot_sptr->size(); ++i )
+      for ( size_t idx = 0; idx < ser_dot_sptr->size(); ++idx )
```
Comment (mwoehlke-kitware) [inline arrows/serialize/protobuf/tests/test_serialize.cxx:427]: "j"?

File: arrows/vxl/bundle_adjust.cxx [hunk 11]
```diff
@@ -256,15 +256,15 @@ ::optimize(camera_map_sptr& cameras,
       }
       super_map_inner_t frame_lm2feature_map;
 
-      for(const track_sptr& t : ftracks)
+      for(const track_sptr& tt : ftracks)
```
Comment (mwoehlke-kitware) [inline arrows/vxl/bundle_adjust.cxx:259]: "ft"? Alternatively, rename the "t" in `SBA_TIMED` instead?

File: sprokit/processes/adapters/embedded_pipeline_extension.h [hunk 12]
```diff
@@ -156,9 +156,9 @@ class embedded_pipeline_extension_registrar
   : public plugin_registrar
 {
 public:
-  embedded_pipeline_extension_registrar( kwiver::vital::plugin_loader& vpl,
-                    const std::string& mod_name )
-    : plugin_registrar( vpl, mod_name )
+  embedded_pipeline_extension_registrar( kwiver::vital::plugin_loader& vpl_,
```
Comment (mwoehlke-kitware) [inline sprokit/processes/adapters/embedded_pipeline_extension.h:159]: I wonder where "vpl" is used? I'd probably go with "loader" instead, and "module_name" (as IIRC was suggested elsewhere).

File: sprokit/processes/adapters/input_adapter_process.cxx [hunk 13]
```diff
@@ -77,16 +77,16 @@ kwiver::adapter::ports_info_t
 input_adapter_process
 ::get_ports()
 {
-  kwiver::adapter::ports_info_t port_info;
+  kwiver::adapter::ports_info_t l_port_info;
```
Comment (mwoehlke-kitware) [inline sprokit/processes/adapters/input_adapter_process.cxx:80]: I'd make this consistent with the change in `output_adapter_process.cxx` and others. (I'd also probably use "π"... ":pie:"... er, "pi" :slightly_smiling_face:.)

(On that note, I do believe C++23 is going to forbid use of emoji as identifier names, so don't use ":pie:" :wink:.)

File: sprokit/processes/adapters/tests/test_non_blocking.cxx [hunk 14]
```diff
@@ -120,7 +120,7 @@ main(int argc, char* argv[])
 }
 
 // ==================================================================
-IMPLEMENT_TEST(non_blocking)
+IMPLEMENT_TEST(nonblocking)
```
Comment (mwoehlke-kitware) [inline sprokit/processes/adapters/tests/test_non_blocking.cxx:123]: Does this expand to "test_non_blocking"? Otherwise it's not obvious where this conflicts? It seems unfortunate to make this name different from the test itself.

File: sprokit/processes/core/serializer_base.cxx [hunk 15]
```diff
@@ -113,11 +113,11 @@ ::base_init()
       elem_spec.m_serializer = std::dynamic_pointer_cast< vital::algo::data_serializer > ( base_nested_algo );
       if ( ! elem_spec.m_serializer )
       {
-        std::stringstream str;
-        str << "Unable to create serializer for type \""
+        std::stringstream sstr;
+        sstr << "Unable to create serializer for type \""
```
Comment (mwoehlke-kitware) [inline sprokit/processes/core/serializer_base.cxx:117]: "str" was not a good name anyway, but "sstr" feels odd; does "ss" work? If not, maybe "stream" or "sstream"?

File: sprokit/src/sprokit/pipeline_util/lex_processor.cxx [hunk 16]
```diff
@@ -414,22 +414,22 @@ ::get_next_token()
       // look for "::"
       if ( c == ':' && n == ':' )
       {
-        token_sptr t = std::make_shared< token > ( m_priv->find_res_word( "::" ), "::" );
-        t->set_location( current_location() );
+        token_sptr tt = std::make_shared< token > ( m_priv->find_res_word( "::" ), "::" );
```
Comment (mwoehlke-kitware) [inline sprokit/src/sprokit/pipeline_util/lex_processor.cxx:417]: "tok"?

File: vital/algo/algorithm_factory.h [hunk 17]
```diff
@@ -214,9 +214,9 @@ class serializer_registrar
    * serializer method. Typical entries could be "protobuf" or "json".
    */
   serializer_registrar( kwiver::vital::plugin_loader& vpl,
-                       const std::string& mod_name,
+                       const std::string& mod,
```
[the remark this answers is not in the document]
Comment (mwoehlke-kitware) [reply vital/algo/algorithm_factory.h:217]: :+1:, also some other places.

File: vital/klv/klv_data.cxx [hunk 18]
```diff
@@ -54,11 +54,11 @@ namespace vital {
 klv_data
 ::klv_data(container_t const& raw_packet,
          std::size_t key_offset, std::size_t key_len,
-         std::size_t m_value_offset, std::size_t value_len)
+         std::size_t value_offset, std::size_t value_len)
```
Comment (mwoehlke-kitware) [inline vital/klv/klv_data.cxx:57]: oh, dear, shame on someone for using an "m_" as a parameter! :slightly_smiling_face:

File: vital/types/feature_track_set.h [hunk 19]
```diff
@@ -65,24 +65,24 @@ class VITAL_EXPORT feature_track_state : public track_state
...
   /// Constructor
-  explicit feature_track_state( frame_id_t frame,
-                                feature_sptr const& feature = nullptr,
-                                descriptor_sptr const& descriptor = nullptr,
-                                bool inlier = false )
-    : track_state{ frame }
-    , feature{ feature }
-    , descriptor{ descriptor }
-    , inlier{ inlier }
+  explicit feature_track_state( frame_id_t p_frame,
```
Comment (mwoehlke-kitware) [inline vital/types/feature_track_set.h:68]: So... this is inconsistent with other places that renamed e.g. "foo" → "foo_". (FWIW, I prefer the "p_foo" convention, i.e. here.)

File: vital/types/geo_polygon.cxx [hunk 20]
```diff
@@ -129,8 +129,8 @@ config_block_get_value_cast( config_block_value_t const& value )
   }
 
   // Set up helper lambda to check for errors
-  auto try_or_die = [&value]( std::istream& s ) {
-    if ( s.fail() ) {
+  auto try_or_die = [&value]( std::istream& s_ ) {
```
Comment (mwoehlke-kitware) [inline vital/types/geo_polygon.cxx:132]: "is"? or "istream"? or "in"?

**Review by mwoehlke-kitware — CHANGES_REQUESTED** [review]

**Comment by linus-sherrill** [comment]
Is this ready to merge. I believe all requested changes have been made.

File: arrows/mvg/algo/initialize_cameras_landmarks.cxx [hunk 21]
```diff
@@ -1,4 +1,4 @@
-/*ckwg +29
+/*cOAkwg +29
```
Comment (mleotta) [inline arrows/mvg/algo/initialize_cameras_landmarks.cxx:1]: looks like some extra character were added by mistake

Comment (linus-sherrill) [reply arrows/mvg/algo/initialize_cameras_landmarks.cxx]: This is now fixed

**Review by mleotta — CHANGES_REQUESTED** [review]
Looks okay to me except for some random characters inserted into a copyright banner comment

**Review by mleotta — APPROVED** [review]
Looks okay now
